diff --git a/clutter/clutter-color.c b/clutter/clutter-color.c
index a062445..583b590 100644
--- a/clutter/clutter-color.c
+++ b/clutter/clutter-color.c
@@ -502,7 +502,7 @@ clutter_color_from_pixel (ClutterColor *dest,
 /**
  * clutter_color_parse:
  * @color: a string specifiying a color (named color or #RRGGBBAA)
- * @dest: (out): return location for a #ClutterColor
+ * @dest: return location for a #ClutterColor
  *
  * Parses a string definition of a color, filling the
  * <structfield>red</structfield>, <structfield>green</structfield>, 
